packages/zcli-apps/src/commands/apps/new.ts

```diff
@@ -87,15 +95,15 @@ export default class New extends Command {
       authorEmail = flags.authorEmail || await cli.prompt('Enter this app authors email')
     }
     const appName = flags.appName || await cli.prompt('Enter a name for this new app')
-    const scaffoldRepo = this.getScaffoldDir(flagScaffold)
-    const scaffoldDir = scaffoldRepo + '-master'
-
-    const scaffoldUrl = `https://codeload.github.com/zendesk/${scaffoldRepo}/zip/master`
+    const scaffoldDir = 'app_scaffolds-master-' + flagScaffold
+    const scaffoldUrl = 'https://codeload.github.com/zendesk/app_scaffolds/zip/master'
 
     try {
-      await this.downloadScaffold(scaffoldUrl)
+      await this.downloadScaffoldsRepo(scaffoldUrl)
```

Comment:
I notice that once a user gets validation error, they are not able to re-create an app anymore. That happens because we download a zip file and don't clean it up immediately.

The clean part you added in this repo, should probably happen soon after we extract the zip file or fail to extract it. I think it will be safer to put that logic inside `extractAllToAsync`  callback inside `downloadScaffoldsRepo` func.

Reply:
I think we need to delete it after we copy the subfolder to a new path, so I'm not sure that I can remove the folder before I extract from it.  I agree that there is an issue here, maybe I can duplicate that line and ensure that it runs in the catch block?

I've added a new line here https://github.com/zendesk/zcli/pull/65/commits/b7969052acf94ca329a457a2e785e0844ffc42b9, I'm not altogether satisfied with this implementation though, maybe there is a cleaner way of doing this?

Reply:
I actually mean't we delete the zip file not the folder here. Rest of the logic seems ok, its just the zip file that remains there when validation error happens causing re-run to fail on a different error (cannot write to zip file, because it already exisits).

Reply:
Basically this,

```
 zip.extractAllToAsync(path.join(process.cwd()), overwrite, (err) => {
           // remove zip file here
           if (err) {
            reject(err)
          }
          resolve()
 }
```